src/main/java/co/inventorsoft/academy/collections/model/Range.java

```diff
@@ -1,60 +1,273 @@
 package co.inventorsoft.academy.collections.model;
 
-import java.util.Collection;
-import java.util.Iterator;
-import java.util.Set;
+import java.util.*;
+import java.util.function.Function;
 
-public class Range<T> implements Set<T> {
+public class Range<T extends Comparable<T>> implements Set<T> {
+
+    private int size = 0;
+    private Node<T> first;
+    private Node<T> last;
+    private Function<T, T> function;
+
+    private static class Node<V> {
+        V item;
+        Node<V> next;
+        Node<V> prev;
+        int id;
+
+        public Node(V item, int id) {
+            this.item = item;
+            this.id = id;
+        }
+
+        Node(Node<V> prev, V element, Node<V> next) {
+            this.item = element;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
+    public static Range<Integer> of(Integer start, Integer end) {
+        Range<Integer> range = new Range<>();
+        if (start < end) {
+            for (Integer i = start; i <= end; i++) {
+                range.add(i);
+            }
+        }
+        return range;
+    }
+
+    public static Range<Float> of(Float start, Float end) {
+        Range<Float> range = new Range<>();
+        for (Float i = start; i <= end; i += 0.1f) {
+            range.add(i);
+        }
+        return range;
+    }
+
+    public static Range<Character> of(Character start, Character end, Function<Character, Character> func) {
+        Range<Character> range = new Range<>();
+        for (Character i = start; i.compareTo(end) <= 0; i = func.apply(i)) {
+            range.add(i);
+        }
+        return range;
+    }
+
+    public boolean add(T t) {
+        boolean flag = false;
+        if (last == null) {
+            addNode(t);
+        } else {
+            if (!contains(t)) {
+                flag = true;
+                addNode(t);
+            }
+        }
+        return flag;
+    }
+
+    private boolean addNode(T t) {
+        final Node<T> l = last;
+        final Node<T> newNode = new Node<>(l, t, null);
+        last = newNode;
+        if (l == null) {
+            first = newNode;
+        } else {
+            l.next = newNode;
+        }
+        size++;
+        return true;
+    }
 
     public int size() {
-        return 0;
+        return size;
     }
 
     public boolean isEmpty() {
-        return false;
+        return size == 0;
     }
 
     public boolean contains(Object o) {
-        return false;
+        boolean flag = false;
+        Iterator<T> iter = iterator();
+        while (iter.hasNext()) {
```

Comment:
Add condition when element when already found element, not to go through all elements.